### PR TITLE
Feat: smt data driven test framework

### PR DESCRIPTION
**Pull request by bvrooman**
Related issues:
- Closes https://github.com/FuelLabs/fuel-merkle/issues/67
- Closes https://github.com/FuelLabs/fuel-merkle/issues/68

This PR adds the initial test execution framework that operates on the data-driven test suite. The framework is defined by a `Test` class that encapsulates a test fixture. The test executes by comparing the expected root defined by the fixture and the root calculated by executing the steps.

The test data is compliant with the test specification outlined in https://github.com/FuelLabs/fuel-merkle/blob/master/docs/test-specs/sparse_merkle_tree_tests.md. Tests are defined as individual fixtures within YAML files.

### CODE REVIEW

File: tests-data/tests-data.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    data: Option<EncodedValue>,
+}
+
+impl Step {
+    pub fn execute(
+        self,
+        tree: &mut MerkleTree<StorageError>,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        match self.action.as_str() {
+            "update" => {
```
Comment (Voxelot) [inline tests-data/tests-data.rs:57]: Should we move these strings into constants?

**Review by Voxelot — APPROVED** [review]
This looks great! I think breaking the fixtures into separate files will help in the long run with being able to track changes easier and make it easier to narrow in on certain kinds of failures.

**Review by adlerjohn — APPROVED** [review]
Looks fine to me. Can you file an issue to remember to move the yaml files to the specs repo sometime, and issues for any other nits?

**Comment by bvrooman** [comment]
Filed the following issues:
- https://github.com/FuelLabs/fuel-merkle/issues/80
- https://github.com/FuelLabs/fuel-merkle/issues/81

**Pull request merged**